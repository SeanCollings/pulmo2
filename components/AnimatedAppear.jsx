import React, { useEffect } from 'react';
import { StyleSheet, Animated, View } from 'react-native';

const AnimateAppear = ({
  style,
  children,
  isVisible,
  startingScale = 0,
  durationScale = 300,
  startingOpacity = 0,
  durationOpacity = 300,
}) => {
  let animatedScale = new Animated.Value(isVisible ? startingScale : 1);
  let animatedOpacity = new Animated.Value(startingOpacity);

  useEffect(() => {
    Animated.parallel([
      Animated.timing(animatedScale, {
        toValue: isVisible ? 1 : 0,
        duration: durationScale,
      }),
      Animated.timing(animatedOpacity, {
        toValue: isVisible ? 1 : 0,
        duration: durationOpacity,
      }),
    ]).start();
  }, [isVisible]);

  const animatedStyleScale = {
    opacity: animatedOpacity,
    transform: [{ scale: animatedScale }],
  };

  if (!isVisible) return null;

  return (
    <View style={{ ...styles.container, ...style }}>
      <Animated.View style={{ ...animatedStyleScale }}>
        {isVisible && children}
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {},
});

export default AnimateAppear;
