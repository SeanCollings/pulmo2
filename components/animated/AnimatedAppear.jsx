import React, { useEffect } from 'react';
import { StyleSheet, Animated, View } from 'react-native';

const AnimateAppear = ({
  style,
  children,
  isVisible,
  startingScale = 0,
  durationScale = 300,
  startingOpacity = 0,
  duration = 300,
}) => {
  let animatedScale = new Animated.Value(isVisible ? startingScale : 1);
  let animatedOpacity = new Animated.Value(startingOpacity);

  useEffect(() => {
    Animated.parallel([
      Animated.timing(animatedScale, {
        toValue: isVisible ? 1 : 0,
        duration: durationScale,
        useNativeDriver: true,
      }),
      Animated.timing(animatedOpacity, {
        toValue: isVisible ? 1 : 0,
        duration: duration,
        useNativeDriver: true,
      }),
    ]).start();
  }, [isVisible]);

  const animatedStyleScale = {
    opacity: animatedOpacity,
    transform: [{ scale: animatedScale }],
  };

  if (!isVisible) return null;

  return (
    <View style={{ ...styles.container, ...style }}>
      <Animated.View style={{ ...animatedStyleScale }}>
        {isVisible && children}
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {},
});

export default AnimateAppear;
