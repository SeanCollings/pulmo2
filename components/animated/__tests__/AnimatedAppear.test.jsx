import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import AnimatedAppear from '../AnimatedAppear';
import { Text } from 'react-native';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('AnimatedAppear - unit test', () => {
  let component;
  const children = <Text>AnimatedAppear</Text>;

  afterEach(() => {
    cleanup();
  });

  test('should render default', async () => {
    await act(async () => {
      component = create(
        <AnimatedAppear isVisible duration={0}>
          {children}
        </AnimatedAppear>
      );
    });

    expect(component).toMatchSnapshot();
  });

  test('should return null if isVisible is false', async () => {
    await act(async () => {
      component = create(
        <AnimatedAppear duration={0}>{children}</AnimatedAppear>
      );
    });

    expect(component).toMatchSnapshot();
  });
});
