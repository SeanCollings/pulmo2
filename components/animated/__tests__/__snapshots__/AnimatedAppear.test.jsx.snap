// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedAppear - unit test should render default 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "opacity": 1,
        "transform": Array [
          Object {
            "scale": 0,
          },
        ],
      }
    }
  >
    <Text>
      AnimatedAppear
    </Text>
  </View>
</View>
`;

exports[`AnimatedAppear - unit test should return null if isVisible is false 1`] = `null`;
